uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Test/Fabricator/TOILRequest.php

```diff
@@ -2,29 +2,42 @@
 
 use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
 use CRM_HRLeaveAndAbsences_BAO_TOILRequest as TOILRequest;
+use CRM_HRLeaveAndAbsences_Test_Fabricator_LeaveBalanceChange as LeaveBalanceChangeFabricator;
 
 class CRM_HRLeaveAndAbsences_Test_Fabricator_TOILRequest {
 
   private static $leaveRequestStatuses;
   private static $toilAmounts;
 
-  public static function fabricate($params) {
+  public static function fabricate($params, $withBalanceChanges = false) {
```

Comment:
I tried to follow same format for LeaveRequest Fabricator which use false as default. Besides I think its better not to create Balance changes by default, it will create unnecessary overhead for tests. The tests that were using the fabricator before, once I moved the balance change logic for the BAO, they started failing so I only added the true parameter only for those that failed. If it were true by default, all the tests would have passed but there would be some that didn't need balance changes to work.